recipes/capstone/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+    topics = ("conan", 'reverse-engineering', 'disassembler', 'security', 'framework', 'arm', 'arm64', 'x86', 'sparc', 'powerpc', 'mips', 'x86-64', 'ethereum', 'systemz', 'webassembly', 'm68k', 'm0s65xx', 'm680x', 'tms320c64x', 'bpf', 'riscv')
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "use_default_alloc": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "use_default_alloc": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake",
+    _cmake = None
+    _archs = ['arm', 'm68k', 'mips', 'ppc', 'sparc', 'sysz', 'xcore', 'x86', 'tms320c64x', 'm680x', 'evm']
+    options.update({a: [True, False] for a in _archs})
+    default_options.update({a: True for a in _archs})
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions['CAPSTONE_BUILD_STATIC'] = not self.options.shared
+        cmake.definitions['CAPSTONE_BUILD_SHARED'] = self.options.shared
+        cmake.definitions['CAPSTONE_BUILD_TESTS'] = False
+        cmake.definitions['CAPSTONE_BUILD_CSTOOL'] = False
+        cmake.definitions['CAPSTONE_ARCHITECUTRE_DEFAULT'] = False
+        for a in self._archs:
+            cmake.definitions['CAPSTONE_%s_SUPPORT' % a.upper()] = self.options.get_safe(a)
+        runtime = self.settings.get_safe("compiler.runtime")
+        if runtime:
+            cmake.definitions['CAPSTONE_BUILD_STATIC_RUNTIME'] = 'MT' in runtime
+        cmake.configure()
+        self._cmake = cmake
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE_LLVM.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Comment:
Maybe a FIXME for missing components?

Should be `capstone-{static,shared|` without a namespace 

https://github.com/aquynh/capstone/blob/71f5c64c43b9868ab08c3a9bad82ba3563e9436d/CMakeLists.txt#L600

Reply:
yes, I am working to automate that part, but it will take some time